rotkehlchen/tests/unit/test_chains_aggregator.py

```diff
@@ -263,3 +265,32 @@ def test_detect_evm_accounts_spam_tx(polygon_pos_manager: 'PolygonPOSManager') -
             db.add_to_ignored_assets(write_cursor=write_cursor, asset=asset)
 
     assert polygon_pos_manager.transactions.address_has_been_spammed(evm_address) is True
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.freeze_time('2024-05-03 10:45:00 GMT')
+@pytest.mark.parametrize('gnosis_manager_connect_at_start', [(GNOSIS_ETHERSCAN_NODE,)])
+@pytest.mark.parametrize('gnosis_accounts', [[make_evm_address()]])  # to connect to nodes
+def test_detect_evm_accounts_spam_tx_gnosis(gnosis_manager: 'GnosisManager') -> None:
+    """
+    Test that an account with only erc20 transfers of spam tokens gets marked as spam
+    and does not get detected as a tracked account in the EVM chain.
+
+    The tested address has received the following spam tokens:
+    eip155:100/erc20:0x8786B9c1a0C676B191284A4Bc7e448321197BB05
+    eip155:100/erc20:0xC1BeC4618B212441E367ED363540D5D665e8e4F0
+    """
+    evm_address = string_to_evm_address('0xF5d90Ac6747CB3352F05BF61f48b991ACeaE28eB')
+
+    for spam_asset in [
+        '0x8786B9c1a0C676B191284A4Bc7e448321197BB05',
+        '0xC1BeC4618B212441E367ED363540D5D665e8e4F0',
+    ]:
+        get_or_create_evm_token(
+            userdb=gnosis_manager.node_inquirer.database,
+            evm_address=string_to_evm_address(spam_asset),
+            chain_id=ChainID.GNOSIS,
+            protocol=SPAM_PROTOCOL,
+        )
+
+    assert gnosis_manager.transactions.address_has_been_spammed(evm_address) is True
```

Comment:
I believe this test is incomplete

1. Add a comment saying that evm_address is the address that only has spam transactions
2. Add a comment stating that the addresses are the addresses of all the unknown spam tokens that the address ahs received
3. You should check that the key in the cache is not present before calling `address_has_been_spammed`
4. Check that after calling `address_has_been_spammed` they key has been set
5. Patch requests.get to ensure assert that the arguments respect the new "from_ts" parameter